src/probes/bert_probe.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * This program is part of the Clear Linux Project
+ *
+ * Copyright 2018 Intel Corporation
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms and conditions of the GNU Lesser General Public License, as
+ * published by the Free Software Foundation; either version 2.1 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+#define _GNU_SOURCE
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <getopt.h>
+#include "common.h"
+#include "telemetry.h"
+#include "nica/b64enc.h"
+
+#include "config.h"
+
+static const char bert_record_file[] = "/sys/firmware/acpi/tables/data/BERT";
+static const char telem_record_class[] = "org.clearlinux/debug/bert";
+static uint32_t severity = 2;
+static uint32_t payload_version = 1;
+
+
+void print_usage(char *prog)
+{
+        printf("%s: Usage\n", prog);
+        printf("  -f,  --config_file    Configuration file. This overides the other parameters\n");
+        printf("  -h,  --help           Display this help message\n");
+        printf("  -V,  --version        Print the program version\n");
+}
+
+static int allocate_payload_buffer(char **payload)
+{
+        int ret = 0;
+
+        *payload = (char *) malloc(MAX_PAYLOAD_SIZE);
+
+        if (*payload == NULL) {
+                goto out1;
+        }
+
+        *payload = memset(*payload, 0, MAX_PAYLOAD_SIZE);
+        ret = 1;
+
+out1:
+        return ret;
+}
+
+int main(int argc, char **argv)
+{
+        struct telem_ref *tm_handle = NULL;
+        char *classification = (char *) telem_record_class;
+        char *payload = NULL;
+
+        // Following vars are for arg parsing.
+        int c;
+        char *cfile = NULL;
+        int opt_index = 0;
+        int ret = 0;
+
+        struct option opts[] = {
+                { "cfile", 1, NULL, 'f' },
```

Comment:
The long option is incorrect. It should be "config-file" (or "config_file" if you want to keep the `--config_file` naming).